src/component/common/ProtectedRoute/ProtectedRoute.jsx

```diff
@@ -1,17 +1,21 @@
-import { Route, Redirect } from 'react-router-dom';
+import { Route, useLocation, useHistory } from 'react-router-dom';
 
 const ProtectedRoute = ({
     component: Component,
     unauthorized,
     renderProps = {},
     ...rest
 }) => {
+    const { pathname } = useLocation();
+    const history = useHistory();
+    const loginLink =
+        pathname.length > 1 ? `/login?redirect=${pathname}` : '/login';
     return (
         <Route
             {...rest}
             render={props => {
                 if (unauthorized) {
-                    return <Redirect to={'/login'} />;
+                    history.push(loginLink);
```

Comment:
Could keep the Redirect component here, so that we return `null` and `replace` the current path:

```
return <Redirect to={loginLink} />;
```